src/lib/prepareOas.ts

```diff
@@ -18,28 +27,94 @@ export default async function prepareOas(path: string, command: 'openapi' | 'val
   let specPath = path;
 
   if (!specPath) {
-    // If the user didn't supply an API specification, let's try to locate what they've got, and validate that. If they
-    // don't have any, let's let the user know how they can get one going.
-    specPath = await new Promise((resolve, reject) => {
-      ['swagger.json', 'swagger.yaml', 'swagger.yml', 'openapi.json', 'openapi.yaml', 'openapi.yml'].forEach(file => {
-        debug(`looking for definition with filename: ${file}`);
-        if (!fs.existsSync(file)) {
-          debug(`${file} not found`);
-          return;
-        }
-
-        info(
-          chalk.yellow(`We found ${file} and are attempting to ${command === 'openapi' ? 'upload' : 'validate'} it.`)
-        );
-        resolve(file);
+    /**
+     * Scans working directory for a potential OpenAPI or Swagger file.
+     * Any files in the `.git` directory or defined in a top-level `.gitignore` file
+     * are skipped.
+     *
+     * A "potential OpenAPI or Swagger file" is defined as a YAML or JSON file
+     * that has an `openapi` or `swagger` property defined at the top-level.
+     *
+     * If multiple potential files are found, the user must select a single file.
+     *
+     * An error is thrown in the following cases:
+     * - if in a CI environment and multiple files are found
+     * - no files are found
+     */
+
+    const fileFindingSpinner = ora({ text: 'Attempting to locate API definitions...', ...oraOptions() }).start();
+
+    const action = command === 'openapi' ? 'upload' : 'validate';
+
+    const ignoreFilter = ignore().add('.git/');
+
+    if (fs.existsSync('.gitignore')) {
+      debug('.gitignore file found, adding to ignore filter');
+      ignoreFilter.add(fs.readFileSync('.gitignore').toString());
+    }
+
+    const jsonAndYamlFiles = readdirRecursive('.', ignoreFilter.createFilter()).filter(
```

Comment:
You might want to integrate the `.gitignore` lookup detection into this method directly so that when it recurs into a new directory it looks for a `.gitignore` there. I don't think we can always assume that `.gitignore` will be in the root.

Reply:
Added here (along with tests): https://github.com/readmeio/rdme/pull/567/commits/ac68c2ab9e4c1ce362ddc66e160b0efa2b4c40c7